src/component/handlers/composition/DraftEditorCompositionHandler.js

```diff
@@ -107,7 +119,16 @@ const DraftEditorCompositionHandler = {
       DraftEditorCompositionHandler.resolveComposition(editor);
       editor._onKeyDown(e);
       return;
+    } else if (!e.nativeEvent.isComposing) {
+      // In some cases, the `compositionend` event may not fire but composition ends anyway.
+      // E.g. in Chrome when you type some text during composition, then undo with CTRL+Z
+      // and exit the composition with ESC
+      // Thus when we receive an event without isComposing flag, we need to end the composition
+      e.preventDefault();
+      DraftEditorCompositionHandler.onCompositionEnd(editor);
+      return;
```

Comment:
Good idea, but I would leave it for later, as DraftJS doesn't use any capture events at all right now, neither does Lexical AFAICT, so it may be a whole new rabbit hole to explore in relation to `contenteditable`